snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -64,16 +77,40 @@ def attempt_map(
 
         return arrayElement(
             expression.alias,
-            Column(None, self.to_table_name, f"{self.to_col_name}.value"),
-            FunctionCall(
+            ColumnExpr(None, self.to_table_name, f"{self.to_col_name}.value"),
+            FunctionCallExpr(
                 None,
                 "indexOf",
                 (
-                    Column(None, self.to_table_name, f"{self.to_col_name}.key"),
+                    ColumnExpr(None, self.to_table_name, f"{self.to_col_name}.key"),
                     expression.key.accept(children_translator),
                 ),
             ),
         )
 
 
+TABLE_TAG_PARAM = "table_name"
+VALUE_COL_TAG_PARAM = "value_column"
+KEY_COL_TAG_PARAM = "key_column"
+KEY_TAG_PARAM = "key"
+tag_pattern = FunctionCall(
+    None,
+    String("arrayElement"),
+    (
+        Column(
+            None,
+            Param(TABLE_TAG_PARAM, AnyOptionalString()),
+            Param(VALUE_COL_TAG_PARAM, Any(str)),
+        ),
+        FunctionCall(
+            None,
+            String("indexOf"),
+            (
+                Column(None, None, Param(KEY_COL_TAG_PARAM, Any(str))),
+                Literal(None, Param(KEY_TAG_PARAM, Any(str))),
+            ),
+        ),
+    ),
+)
```

Comment:
I think we should rename the pattern classes not to conflict with the AST classes (or having helper function to create them so that they are lower case). They are almost always being used in the same file which means we must always alias one of the two in the import. This is fairly inconvenient.

Reply:
My thoughts: I think either appending `Matcher` to these classes is fine (e.g. `StringMatcher`) or just using `from snuba.query import matchers` and qualifying them as `matchers.String`, etc. would be fine.